commands/SearchController.php

```diff
@@ -0,0 +1,358 @@
+
+
+    public function options($actionID)
+    {
+        if ($actionID === 'ranking') {
+            return array_merge(parent::options($actionID), ['progress']);
+        }
+        return parent::options($actionID);
+    }
+
+    public function actionIndex()
+    {
+        $this->stdout("Search settings\n\n", Console::BOLD);
+
+        /** @var $es Connection */
+        $es = Yii::$app->elasticsearch;
+
+        $this->stdout('Elasticsearch config: ' . print_r($es->nodes, true));
+
+        $this->stdout('Elasticsearch indices: ' . print_r(array_keys($es->createCommand()->getIndexStats()['indices']), true));
+
+
+    }
+
+    /**
+     * Test how good common search queries work.
+     */
+    public function actionTest()
+    {
+        $queries = [
+            'Yii',
+            'Install Yii',
+            'How the hell do I isntall Yii?', // typo intentionally added
+            'Active Record',
+            'ActiveRecord',
+            'Dataprovider',
+            'logging',
+            'logs',
+            'asset management',
+            'theme yii',
+            'how to get started?',
+            'grid view',
+            'gridview',
+            'cgridview',
+        ];
+
+        foreach ($queries as $query) {
+            $this->stdout("$query\n", Console::BOLD);
+            $q = SearchActiveRecord::search($query);
+            foreach($q->limit(5)->all() as $item) {
+                $this->stdout(" - ");
+                $this->stdout("$item->type ", Console::FG_YELLOW);
+                if ($item->hasAttribute('version')) {
+                    $this->stdout("$item->version ", Console::FG_BLUE);
+                }
+                if ($item->hasAttribute('language')) {
+                    $this->stdout("$item->language ", Console::FG_GREEN);
+                }
+                $this->stdout($item->getTitle() . "\n");
+            }
+        }
+    }
+
+    /**
+     * Drop all search indices
+     */
+    public function actionClear()
+    {
+        // TODO set index analyzer:
+        // https://www.elastic.co/guide/en/elasticsearch/reference/5.6/indices-update-settings.html#update-settings-analysis
+        // https://www.elastic.co/guide/en/elasticsearch/reference/5.6/indices-analyze.html
+
+        $index = SearchActiveRecord::index();
+        $command = SearchActiveRecord::getDb()->createCommand();
+        if ($command->indexExists($index)) {
+            $command->deleteIndex($index);
+        }
+        foreach (SearchActiveRecord::$languages as $lang => $analyzer) {
+            if ($command->indexExists("$index-$lang")) {
+                $command->deleteIndex("$index-$lang");
+            }
+        }
+    }
+
+    /**
+     * Rebuild search index.
+     */
+    public function actionRebuild(array $items = ['guide', 'api', 'extension', 'wiki', 'news'])
+    {
+        foreach($items as $item) {
+            switch($item) {
+                case 'guide':
+                    // SearchGuideSection::deleteAll();
```

Comment:
Why isn't it deleted here?

Reply:
items will be overwritten when reindexing. This way they still appear in the search while reindexing process is running. Will clean this up.